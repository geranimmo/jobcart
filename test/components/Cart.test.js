import React from 'react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { mount, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import expect from 'expect';
import renderer from 'react-test-renderer';
import ConnectedCart, { Cart } from '../../src/components/cart';
import ListPackages from '../../src/assets/json/packages__.json';
import {
	getTotalCost,
	getTotalDiscount
} from '../../src/components/actions';

describe('>>> C A R T ---- Test & Snapshot <<<', () => {
	const mockStore = configureMockStore([thunk]);
	let wrapper, store;

	const propCart = {
		ShoppingCart: [
			{
				id: 'Classic',
				buy_time: 1537425742947,
				package_id: "Classic",
				package_name: "Classic",
				package_recommend: false,
				package_description: "Classic Ad Lorem Ipsum is simply dummy text of the printing and typesetting",
				package_price: 269.99,
				package_image: "/assets/images/employeer-image-sample.jpg"
			}, {
				id: 'Premium',
				buy_time: 1537425742948,
				package_id: "Classic",
				package_name: "Classic",
				package_recommend: false,
				package_description: "Classic Ad Lorem Ipsum is simply dummy text of the printing and typesetting",
				package_price: 269.99,
				package_image: "/assets/images/employeer-image-sample.jpg"
			}
		],
		Profile: {
			client_id: 1,
			first_name: "Default",
			last_name: "Client",
			business_name: "Default",
			phone_number: "+612529291002",
			client_logo: "client-default-logo.png",
			client_special: []
		},
		UserLogin: true,
		Packages: ListPackages,
		id: 'shopping__cart__wrapper',
		getTotalCost,
		getTotalDiscount
	};

	beforeEach(() => {
		store = mockStore(propCart);
		wrapper = mount(
			<Provider store={store}>
				<Router>
					<ConnectedCart {...propCart}/>
				</Router>
			</Provider>
		);
	});

	it('+++ Render the ConnectedCart component', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('+++ Contains props that assigned to the ConnectedCart component', () => {
		expect(wrapper.find(`main`).prop('id')).toEqual(propCart.id);
	});

	it('+++ Capturing Snapshot of ConnectedCart component +++', () => {
		const renderedValue = renderer.create(
			<Provider store={store}>
				<Router>
					<ConnectedCart {...propCart}/>
				</Router>
			</Provider>
		).toJSON();
		
		expect(renderedValue).toMatchSnapshot();
	});

	it('+++ Simulate Click Remove From Cart in Packages +++', () => {
		const removeFromCart = jest.fn();
		
		expect(
			wrapper
				.find('div#content__scroller')
				.find('.list__0')
				.find('.cart__delete')
				.length
		).toEqual(1);
		
		wrapper
			.find('div#content__scroller')
			.find('.list__0')
			.find('.cart__delete')
			.simulate('click');

		expect(removeFromCart).toEqual(expect.any(Function));
	});

	it('+++ Simulate componentWillReceiveProps +++', () => {
		const history = { push: jest.fn() };
		const getListPackage = jest.fn();
		const wrappers = shallow(
			<Cart
				{...propCart}
				history={history}
				getListPackage={getListPackage}
			/>
		);
		const showCartSpy = jest.spyOn(wrappers.instance(), 'createDataSource');

		wrappers.setProps({ UserLogin: false });
		wrappers.instance().forceUpdate();
		wrappers.update();
		expect(showCartSpy.mock.calls.length).toBe(1);
	});

	it('+++ Should call handle scroll function when scroll +++', () => {
		const handleHeaderOnScroll = jest.fn();

		wrapper.find('div#content__scroller').simulate('scroll', { deltaY: 50 });
		expect(handleHeaderOnScroll).toEqual(expect.any(Function));
	});
});