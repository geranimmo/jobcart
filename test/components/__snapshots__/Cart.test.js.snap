// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> C A R T ---- Test & Snapshot <<< +++ Capturing Snapshot of ConnectedCart component +++ 1`] = `
<div
  className="main__container"
>
  <header
    className="header__active"
    id="header"
  >
    <nav
      className="nav__left"
    >
      <a
        aria-current={null}
        href="/home"
        onClick={[Function]}
      >
        <div
          className="seek__logo"
        />
      </a>
      <div
        className="client__logo__wrapper "
      >
        <img
          alt="Default"
          src="client-default-logo.png"
        />
      </div>
    </nav>
    <nav
      className="nav__right "
    >
      <div
        className="menu__btn__icon hidden"
        id="openCartView"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24 3l-.743 2h-1.929l-3.474 12h-13.239l-4.615-11h16.812l-.564 2h-13.24l2.937 7h10.428l3.432-12h4.195zm-15.5 15c-.828 0-1.5.672-1.5 1.5 0 .829.672 1.5 1.5 1.5s1.5-.671 1.5-1.5c0-.828-.672-1.5-1.5-1.5zm6.9-7-1.9 7c-.828 0-1.5.671-1.5 1.5s.672 1.5 1.5 1.5 1.5-.671 1.5-1.5c0-.828-.672-1.5-1.5-1.5z"
          />
        </svg>
        <div
          className="cart__notif "
        >
          2
        </div>
      </div>
      <div
        className="menu__btn__icon "
        id="closeCartView"
        onClick={[Function]}
      >
        <svg
          className="icon__close"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M23.954 21.03l-9.184-9.095 9.092-9.174-2.832-2.807-9.09 9.179-9.176-9.088-2.81 2.81 9.186 9.105-9.095 9.184 2.81 2.81 9.112-9.192 9.18 9.1z"
          />
        </svg>
      </div>
    </nav>
  </header>
  <main
    className="cart__component"
    id="shopping__cart__wrapper"
  >
    <div
      id="content__scroller"
      onScroll={[Function]}
    >
      <div
        className="cart__empty hidden"
      >
        Your Cart Still Empty
      </div>
      <div
        className="wrapper__list0"
        id="Cart__Classic__0__1537425742947"
      >
        <div
          className="cart__list__wrapper"
        >
          <div
            className="cart__info__wrapper"
          >
            <div
              className="cart__info"
            >
              <h1>
                Classic
                <label
                  className="info__recommended hidden"
                >
                  RECOMMENDED
                </label>
              </h1>
              <h2>
                Classic Ad Lorem Ipsum is simply dummy text of the printing and typesetting
              </h2>
              <span
                className="info__datetime"
              >
                09/20/2018, 1:42:22 PM
              </span>
              <span
                className="info__price"
              >
                $
                269.99
              </span>
            </div>
          </div>
          <div
            className="cart__delete"
            id="Classic__1537425742947"
            onClick={[Function]}
          >
            <svg
              clipRule="evenodd"
              fillRule="evenodd"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 24h-14c-1.104 0-2-.896-2-2v-17h-1v-2h6v-1.5c0-.827.673-1.5 1.5-1.5h5c.825 0 1.5.671 1.5 1.5v1.5h6v2h-1v17c0 1.104-.896 2-2 2zm0-19h-14v16.5c0 .276.224.5.5.5h13c.276 0 .5-.224.5-.5v-16.5zm-9 4c0-.552-.448-1-1-1s-1 .448-1 1v9c0 .552.448 1 1 1s1-.448 1-1v-9zm6 0c0-.552-.448-1-1-1s-1 .448-1 1v9c0 .552.448 1 1 1s1-.448 1-1v-9zm-2-7h-4v1h4v-1z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="wrapper__list1"
        id="Cart__Premium__1__1537425742948"
      >
        <div
          className="cart__list__wrapper"
        >
          <div
            className="cart__info__wrapper"
          >
            <div
              className="cart__info"
            >
              <h1>
                Classic
                <label
                  className="info__recommended hidden"
                >
                  RECOMMENDED
                </label>
              </h1>
              <h2>
                Classic Ad Lorem Ipsum is simply dummy text of the printing and typesetting
              </h2>
              <span
                className="info__datetime"
              >
                09/20/2018, 1:42:22 PM
              </span>
              <span
                className="info__price"
              >
                $
                269.99
              </span>
            </div>
          </div>
          <div
            className="cart__delete"
            id="Premium__1537425742948"
            onClick={[Function]}
          >
            <svg
              clipRule="evenodd"
              fillRule="evenodd"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 24h-14c-1.104 0-2-.896-2-2v-17h-1v-2h6v-1.5c0-.827.673-1.5 1.5-1.5h5c.825 0 1.5.671 1.5 1.5v1.5h6v2h-1v17c0 1.104-.896 2-2 2zm0-19h-14v16.5c0 .276.224.5.5.5h13c.276 0 .5-.224.5-.5v-16.5zm-9 4c0-.552-.448-1-1-1s-1 .448-1 1v9c0 .552.448 1 1 1s1-.448 1-1v-9zm6 0c0-.552-.448-1-1-1s-1 .448-1 1v9c0 .552.448 1 1 1s1-.448 1-1v-9zm-2-7h-4v1h4v-1z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </main>
  <div
    className="cart__footer"
  >
    <div>
      <span>
        <label>
          Total Cost:
        </label>
        $
        NaN
      </span>
      <button
        className="button__checkout "
        type="button"
        value="Checkout (2)"
      >
        Checkout (2)
      </button>
    </div>
  </div>
</div>
`;
